### ISSUE TITLE
[workspace - test] `cargo test` should output a summary after running all tests in a workspace

### ISSUE THREAD

**Issue by robinmoussu** [rust-lang/cargo#8414]
**Describe the problem you are trying to solve**

In a "standard" crate, if you run `cargo test`, and everything is good, you can look at the last line of the output. In a crate that uses workspace, the output is much more messy, since each sub-crate will report their status, and you can't see immediately if everything went fine.

**Describe the solution you'd like**

In addition to the current output, if the current crate uses workspace, a global summary should be added at the end of the logs. Something like:

```
   Summary (in green)

test result: ok. 54 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out
```

**Comment by avl** [comment]
This would be very nice. I have a largish project at work with a lot of crates. The output from 

'cargo test --workspace'

is 268 lines long. I just realized I've had a bug on a working branch, detected by tests for several days, without realizing. The failing tests simply scrolled past too fast for me to be able to read see it.

**Comment by FredTheDino** [comment]
Gonna go ahead and bump this! I've had similar problems to avl, not to mention that the output is quite noisy.

I took a quick look at the source, and this looks like a hard feature to implement. Since it looked like each sub-crate was compiled and run individually. This lead me to the conclusion that the easiest way might be to do a cargo-plugin that parses the output from `cargo test`.

I'm in a fixing mood and might just go ahead and write this thing - unless someone else sees an easier or better way?

**Comment by weihanglo** [comment]
From <https://github.com/rust-lang/cargo/issues/11945#issuecomment-1516150656>:

> Feel like this is also blocked on libtest's JSON output. Cargo needs a reliable way to collect such data. See https://github.com/rust-lang/cargo/issues/11875#issuecomment-1479603968 for more. In the meantime, personally I'd recommend [nextest](https://github.com/nextest-rs/nextest), which has a handful of goodies for testing user experience.

**Comment by epage** [comment]
This reads like its a duplicate of #4324, closing in favor of that.  If there is a reason we need a unique issue for this, let us know!
